k9mail/src/main/java/com/fsck/k9/view/MessageHeader.java

```diff
@@ -154,10 +185,31 @@ private void onAddSenderToContacts() {
         }
     }
 
+    private void onAddAddressesToClipboard(Address[] addresses){
+        if(mMessage != null){
+            try{
+                String addressesToCopy = "";
+                for(Address addressTemp : addresses){
+                    addressesToCopy += addressTemp.getAddress() + " ";
+                }
+                addressesToCopy = addressesToCopy.substring(0,addressesToCopy.length()-1);
+                ClipboardManager.getInstance(mContext).setText("addresses", addressesToCopy);
+                Toast.makeText(mContext, R.string.copy_address_to_clipboard, Toast.LENGTH_LONG).show();
+            }
+            catch (Exception e){
```

Comment:
Catching all exceptions is a bad habit that quickly gets out of hand. K-9 Mail is especially guilty of that. Let's try not to make it worse.
We want unexpected exceptions to crash the app so we can learn about them and then deal with those exceptions explicitly.